jena-core/src/test/java/org/apache/jena/util/TestPrefixMappingUtils.java

```diff
@@ -141,14 +146,125 @@ static int size(PrefixMapping pmap) {
         Graph graph = create(data) ;
         PrefixMapping pmap = PrefixMappingUtils.calcInUsePrefixMapping(graph) ;
         PrefixMapping pmapExpected = new PrefixMappingImpl() ;
-     	pmapExpected.setNsPrefix("", "http://example/") ;
-		pmapExpected.setNsPrefix("ex", "http://example/ex#") ;
-		pmapExpected.setNsPrefix("indirect", "urn:foo:") ;
-		pmapExpected.setNsPrefix("ns", "http://host/ns") ;
-		pmapExpected.setNsPrefix("ns1", "http://host/ns1") ;
-		pmapExpected.setNsPrefix("indirectx", "urn:x:") ;
-		//print("Expected:", pmapExpected) ;
-		//print("Got:", pmap) ;
+        pmapExpected.setNsPrefix("", "http://example/") ;
+        pmapExpected.setNsPrefix("ex", "http://example/ex#") ;
+        pmapExpected.setNsPrefix("indirect", "urn:foo:") ;
+        pmapExpected.setNsPrefix("ns", "http://host/ns") ;
+        pmapExpected.setNsPrefix("ns1", "http://host/ns1") ;
+        pmapExpected.setNsPrefix("indirectx", "urn:x:") ;
+        //print("Expected:", pmapExpected) ;
+        //print("Got:", pmap) ;
+        Assert.assertTrue(sameMapping(pmapExpected, pmap)) ;
+        Assert.assertTrue(pmap.getNsPrefixURI("notinuse") == null) ;
+    }
+
+    @Test public void prefixesTTL1() {
+        // All prefixes used.
+        // Note: XSD as well.
+        String data1 = StrUtils.strjoinNL
+            ("@prefix : <http://example/> ." ,
+             "@prefix ex: <http://example/ex#> ." ,
+             "@prefix xsd: <"+XSD.getURI()+"> .",
+             "" ,
+             ":s1 :p :x1 ." ,
+             ":s1 ex:p :x1 .",
+             ":s1 ex:p 1 ."
+             ) ;
+        Graph graph1 = create(data1) ;
+        PrefixMapping pmap = PrefixMappingUtils.calcInUsePrefixMappingTTL(graph1) ;
+        PrefixMapping pmapExpected = graph1.getPrefixMapping() ;
+        Assert.assertEquals(3, size(pmap)) ;
+        Assert.assertEquals(pmapExpected, pmap) ; 
+    }
+    
+    @Test public void prefixesTTL2() {
+        // Some prefixes used
+        String data2 = StrUtils.strjoinNL
+            ("@prefix : <http://example/> ." ,
+             "@prefix ex: <http://example/ex#> ." ,
+             "@prefix notinuse: <http://example/whatever/> ." ,
+             "" ,
+             ":s1 :p :x1 ." ,
+             ":s1 ex:p :x1 ."
+             ) ;
+        
+        Graph graph1 = create(data2) ;
+        PrefixMapping pmap = PrefixMappingUtils.calcInUsePrefixMappingTTL(graph1) ;
+        PrefixMapping pmapExpected = new PrefixMappingImpl() ;
+        pmapExpected.setNsPrefix("", "http://example/") ;
+        pmapExpected.setNsPrefix("ex", "http://example/ex#") ;
+        Assert.assertEquals(2, size(pmap)) ;
+        Assert.assertTrue(sameMapping(pmapExpected, pmap)) ;
+        Assert.assertTrue(pmap.getNsPrefixURI("notinuse") == null) ;
```

Comment:
Tests confirming the prefixes are available after the calc methods are called :+1: